shopify/resources/product.py

```diff
@@ -30,3 +30,16 @@ def remove_from_collection(self, collection):
     def add_variant(self, variant):
         variant.attributes['product_id'] = self.id
         return variant.save()
+
+    def save(self):
+        # github issue 347
+        # todo: how to get the api version without split & strip
+        api_version = ShopifyResource._site.split('/')[-1].strip('-')
```

Comment:
I believe you should be able to access this directly with `ShopifyResource._version`

Reply:
I have tried, but `ShopifyResource._version` has not value.

Reply:
What about `self.version`?  This should all be available on ShopifyResources https://github.com/Shopify/shopify_python_api/blob/master/shopify/base.py#L92-L155

(for some reason version is duplicated as well 🤦‍♂ )

Reply:
@tanema nop, _ShopifyResource.version_ not return value, and _self.version_ does not exist.

Reply:
sorry I should have been more clear. I mean you can actually access it from self. I just tested this myself and it works.

```python
api_version = self.version.strip('-') # this will convert 2020-01 to 202001 which it looks like you are comparing.
```

Also, both the comparison here and the URL stripping do not handle `unstable`

```python
    def save(self):
        start_api_version = '201910'
        if self.version.strip('-') >= start_api_version and self.version != 'unstable':
            for variant in self.variants:
                if 'inventory_quantity' in variant.attributes:
                    del variant.attributes['inventory_quantity']
                if 'old_inventory_quantity' in variant.attributes:
                    del variant.attributes['old_inventory_quantity']
        return super(ShopifyResource, self).save()
```

Reply:
When I call on my script:

>         success = shopify_product.save()

And intercept with pdb, I dont have self.version:
```
ShopifyAPI-7.0.1-py3.7.egg/shopify/resources/product.py(40)save()
-> if api_version >= start_api_version:
(Pdb) self.version
*** AttributeError: version

```

Reply:
Sorry, still coming to grips with this codebase. It is actually referenced from `_threadlocal` to keep thread saftey.

```
    def save(self):
        api_version = self._threadlocal.version
        start_api_version = '201910'
        if api_version.strip('-') >= start_api_version and api_version != 'unstable':
            for variant in self.variants:
                if 'inventory_quantity' in variant.attributes:
                    del variant.attributes['inventory_quantity']
                if 'old_inventory_quantity' in variant.attributes:
                    del variant.attributes['old_inventory_quantity']
        return super(ShopifyResource, self).save()
```

Reply:
Also no value. I think this happens when you dont use session (it's my case). I deleted some repeated lines in _base.py_ and updated getter _get_version_ to retrieve the api version from the path when no value in _version_ or __version_